src/randomizer.py

```diff
@@ -2080,6 +2103,85 @@ def visit_Decl(self, decl: ca.Decl) -> None:
     ast_util.insert_decl(fn, var, type, random)
 
 
+def perm_inline_get_structmember(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Creates an inline function for accessing a struct member."""
+
+    typemap = build_typemap(ast)
+
+    cands: List[ca.StructRef] = []
+
+    class Visitor(ca.NodeVisitor):
+        def visit_StructRef(self, node: ca.StructRef) -> None:
+            if (
+                region.contains_node(node)
+                and isinstance(node.name, ca.ID)
+                and not ast_util.is_assignment_lvalue(fn.body, node)
+            ):
+                cands.append(node)
+            self.generic_visit(node)
+
+    Visitor().visit(fn.body)
+    ensure(cands)
+
+    cand = random.choice(cands)
+
+    parentOp: Optional[ca.UnaryOp] = ast_util.find_parent_addressOP(fn.body, cand)
+
+    member_type: SimpleType = decayed_expr_type(cand, typemap)
+    member_name = cand.field.name
+
+    new_inline_fn_name = get_noncolliding_fn_name(ast, f"get_{member_name}")
+
+    # Replace the StructRef with a FuncCall to the new inline yet to be created
+    replace_node(
+        fn.body,
+        cand if not parentOp else parentOp,
+        ca.FuncCall(ca.ID(new_inline_fn_name), ca.ExprList([cand.name])),
+    )
+
+    # Now create the new inline function definition
+    arg_type: SimpleType = decayed_expr_type(cand.name, typemap)
+    assert isinstance(arg_type.type, ca.TypeDecl)
+    assert arg_type.type.declname
+
+    fn_decl = ca.Decl(
+        name=new_inline_fn_name,
+        quals=[],
+        align=[],
+        storage=[],
+        funcspec=["inline"],
+        type=ca.FuncDecl(
+            args=ca.ParamList(
+                [ca.Typename(name=None, quals=[], align=[], type=arg_type)]
```

Comment:
You want a Decl for the param; a Typename is only for the case where the param does not have a name

Reply:
okay sure, I changed it... but I don't understand, typename worked and does have the name

Reply:
Decl, just seems like a lot for a param

```
    theParam = ca.Decl(
        name=None,
        quals=[],
        align=[],
        storage=[],
        funcspec=[],
        type=arg_type,
        init=None,
        bitsize=None,
    )
```

Reply:
There may be further randomization passes, and if we generate invalid AST's (that just happen to stringify to the right thing) they may break. To combat the verbosity, you could create a function `make_decl(name, type)` (which also calls `set_decl_name(decl)`, and which `insert_decl` can make use of).